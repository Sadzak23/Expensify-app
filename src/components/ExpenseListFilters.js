import React from "react";
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from "../actions/filters";
import 'react-dates/initialize';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState({ calendarFocused });
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
  };
  onSortByChange = (e) => {
    if (e.target.value === "amount") {
      this.props.sortByAmount()
    } else if (e.target.value === "date") {
      this.props.sortByDate()
    }
  };
  render() {
    return (
      <div className="content-container">
        <div className="item_group">
          <div className="item_group__item">
            <input
              className="text-input"
              type="text"
              placeholder="Search expenses"
              value={this.props.filters.text}
              onChange={this.onTextChange} />
          </div>
          <div className="item_group__item">
            <select
              className="select"
              value={this.props.filters.value}
              onChange={this.onSortByChange}>
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="item_group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              startDateId="start-date_id"
              endDate={this.props.filters.endDate}
              endDateId="end-date_id"
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              firstDayOfWeek={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate())

});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)