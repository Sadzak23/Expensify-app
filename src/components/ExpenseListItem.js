import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ dispatch, description, amount, createdAt, id }) => (
  <Link className="expense-list-item" to={`/edit/${id}`}>
    <div className="expense-list-exp">
      <h3>{description}</h3>
      <p>{moment(createdAt).format("DD. MMMM YYYY.")}</p>
    </div>
      <h3 className="expense-list-amount">{numeral(amount).format('$0,0.00')}</h3>
  </Link>
);

export default ExpenseListItem;