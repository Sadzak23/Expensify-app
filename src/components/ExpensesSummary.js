import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const formattedTotal = numeral(expensesTotal).format('$0,0.00');
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedTotal}</span></h1>
        <Link className="big-button" to="/create">Add Expense</Link>
      </div>
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
};

export default connect(mapStateToProps)(ExpensesSummary)