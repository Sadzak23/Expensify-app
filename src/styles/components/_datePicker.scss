.DateInput {
  padding-right: 22px;
  width: 110px;
}