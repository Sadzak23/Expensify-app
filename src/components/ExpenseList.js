import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
  <div className="content-container">
    <div>
      <div className="expense-list-header">
        <p className="show-mobile">Expenses</p>
        <p className="show-desktop">Expense</p>
        <p className="show-desktop">Amount</p>
      </div>
      <div className="expense-list-body">
        {
          props.expenses.length === 0 ? (
            <p className="expense-list-no-exp">No expenses</p>
          ) : (
              props.expenses.map(expense =>
                (<ExpenseListItem
                  key={expense.id}
                  {...expense}
                />))
            )
        }
      </div>
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList)
