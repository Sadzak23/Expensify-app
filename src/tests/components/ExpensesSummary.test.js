import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test("Should render ExpensesSummary with 1 expense correctly", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1520} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render ExpensesSummary with multiple expense correctly", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={expenses.length} expensesTotal={6020} />);
  expect(wrapper).toMatchSnapshot();
});