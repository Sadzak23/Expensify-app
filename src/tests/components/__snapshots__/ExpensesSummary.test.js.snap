// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1,520.00
      </span>
    </h1>
    <Link
      className="big-button"
      replace={false}
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;

exports[`Should render ExpensesSummary with multiple expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $6,020.00
      </span>
    </h1>
    <Link
      className="big-button"
      replace={false}
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;
